sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/LN0Adapter.java

```diff
@@ -51,4 +52,12 @@ protected  boolean matchesDataAttributes(String dataAttribute){
                 currentElem.getSampledValueControl().stream().anyMatch(smp -> smp.getName().equals(dataAttribute)) ||
                 currentElem.getGSEControl().stream().anyMatch(gse -> gse.getName().equals(dataAttribute));
     }
+
+    @Override
+    public void removeAllControlBlocksAndDatasets() {
+        super.removeAllControlBlocksAndDatasets();
+        // use reflection because JAXB generated class do not have setters for List fields
+        Utils.setField(currentElem, "gseControl", null);
```

Comment:
Why not call method clear on list?

Reply:
Thanks to your comment, I found a way to configure xjc plugin to generate `isSet___` methods for each fields. So We no longer requires reflection.
I removed `Utils.setField` and replaced it with `isSet___` calls where needed.